apps/condo/domains/marketplace/hooks/useMarketplaceServicesFilters.tsx

```diff
@@ -1,40 +1,160 @@
     const intl = useIntl()
-    const EnterStatusMessage = intl.formatMessage({ id: 'pages.condo.payments.enterStatus' })
-    const StatusTitle = intl.formatMessage({ id: 'Status' })
+    const SkuMessage = intl.formatMessage({ id: 'pages.condo.marketplace.services.table.sku' })
+    const NameMessage = intl.formatMessage({ id: 'pages.condo.marketplace.services.table.name' })
+    const CategoryMessage = intl.formatMessage({ id: 'pages.condo.marketplace.services.table.category' })
+    const EnterAddressMessage = intl.formatMessage({ id: 'pages.condo.meter.EnterAddress' })
+    const AllPropertiesMessage = intl.formatMessage({ id: 'pages.condo.marketplace.services.table.allProperties' })
 
-    const userOrganization = useOrganization()
-    const userOrganizationId = get(userOrganization, ['organization', 'id'])
+    const router = useRouter()
+    const { organization } = useOrganization()
+    const userOrganizationId = get(organization, 'id')
+
+    const { count: propertiesCount } = Property.useCount({
+        where: {
+            organization: { id: userOrganizationId },
+        },
+    })
+    const isSingleProperty = propertiesCount === 1
+
+    const { objs: properties } = Property.useObjects({
+        where: {
+            organization: { id: userOrganizationId },
+        },
+    }, { skip: !isSingleProperty })
+
+    const filtersFromQuery = useMemo(() => getFiltersFromQuery(router.query), [router.query])
+    const searchValueFromQuery = get(filtersFromQuery, 'search')
+    const propertyValueFromQuery = get(filtersFromQuery, 'property')
+
+    const propertyWhere = {}
+    if (searchValueFromQuery) {
+        propertyWhere['address_contains_i'] = searchValueFromQuery
+    }
+    if (propertyValueFromQuery) {
+        propertyWhere['id_in'] = propertyValueFromQuery
+    }
+
+    const isAllPropertiesSearching = isSingleProperty ? get(properties, '0.address', '').includes(searchValueFromQuery) :
+        AllPropertiesMessage.includes(searchValueFromQuery)
+    const isAllPropertiesSelected = get(propertyValueFromQuery, 'length') === propertiesCount
+
+    const { objs: marketPriceScopes } = MarketPriceScope.useAllObjects({
+        where: {
+            OR: [
+                { AND: [ { property: propertyWhere } ] },
+                (isAllPropertiesSearching || isAllPropertiesSelected) && { property_is_null: true },
+            ].filter(Boolean),
+        },
+    }, { skip: isEmpty(searchValueFromQuery) && isEmpty(propertyValueFromQuery) })
+
+    const searchAddressFilter = useCallback(() => {
+        if (isEmpty(searchValueFromQuery)) {
+            return
+        }
+
+        const marketItemIdsFromScopes = uniq(marketPriceScopes.map(scope => get(scope, 'marketItemPrice.marketItem.id')))
+
+        return { id_in: marketItemIdsFromScopes }
+    }, [marketPriceScopes, searchValueFromQuery])
+
+    const propertyFilter = useCallback(() => {
+        if (isEmpty(propertyValueFromQuery)) {
```

Comment:
Is it ok that you check for `propertyValueFromQuery` and not use it?
The same is true within `searchAddressFilter` above.

Reply:
Yes. `propertyValueFromQuery` and `searchAddressFilter` used in `marketPriceScopes` where filter, and from `marketPriceScopes` we take out `marketItem` ids for filter `marketItem`. If there is no `propertyValueFromQuery` in query, then no need to filter `marketItems` by `marketPriceScopes` (same for `searchAddressFilter`)